reports/appendix.tex

```diff
@@ -61,7 +63,10 @@ \subsection{Proofs}
 % we do not need to consider the second branch of the definition to compute $\op(x)$. Thus, we do not need to recursively compute $\op(v)$ for any $v$, and $h(V(x))$ is empty.
 % \end{proof}
 
-We define an auxiliary function $\alpha$ that assigns every function of the type $a \to Set(a)$ a value in $\nat \cup \{-\infty\}$, defined as:
+We define an auxiliary function $\alpha$ that assigns every function of the type $a \to Set(a)$ a value in $\nat \cup \{-\infty\}$, defined as
+\tth{Does sup refer to supremum?}
+\tth{Under what condition $\alpha(f) = -\infty$? Does there exist any case where $\alpha(f) = \infty$?}
```

Comment:
By definition, the supremum is the _lowest upper bound_. For the case where a set is empty, every value `v` is an upper bound by the vacuously true formula
```
\A x \in {}: v >= x 
```
so for the empty set the _lowest_ upper bound does not exist (cannonically denoted `sup {} = -inf`). Conversely, if no upper bound existed at all (i.e. if the set contained arbitrarily large elements) we would say `sup S = inf`. Observe that this is only possible for infinite sets, which we are not dealing with, so `alpha(f) = inf` is impossible in our setting.